universe/syncer.go

```diff
@@ -116,9 +122,31 @@ func (s *SimpleSyncer) executeSync(ctx context.Context, diffEngine DiffEngine,
 	// remote servers.
 	case globalInsertEnabled:
 		log.Infof("Fetching all roots for remote Universe server...")
-		targetRoots, err = diffEngine.RootNodes(ctx, false)
-		if err != nil {
-			return nil, err
+
+		offset := int32(0)
+
+		for {
```

Comment:
Ah cool, didn't realize we went ahead to also update the syncers in this version. We should make sure that once we extend past the page size, this logic works properly. The load tests can be a good sanity check, but eventually I think we'll want to add some unit tests here (as the`diffEngine`) is already abstracted to ensure proper termination.

Reply:
Yeah, as is with the itests rn, we'll only have a single page of assets worth. If we extract this out into a diff function, then we can write tests for the pagination/termination logic, and we should also eventually add itests and unit tests at the DB level for the behavior as well.